compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -808,14 +801,34 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
 
         if let ObligationCauseCode::ImplDerivedObligation(obligation) = &*code {
             let expected_trait_ref = obligation.parent_trait_ref;
-            let new_imm_trait_ref = poly_trait_ref
+            let found_ty = expected_trait_ref.self_ty().skip_binder();
+            let imm_borrowed_found_ty = self.tcx.mk_imm_ref(self.tcx.lifetimes.re_static, found_ty);
```

Comment:
Why can we use a static lifetime here? Is it just that the actual lifetime is irrelevant for the diagnostics and we're fine with any lifetime?

Reply:
Mostly, yeah. This will indeed lead to wrong diagnostics in some cases, but changing this to an inference variable would yield more diagnostics that are not necessarily any more right. The "correct" solution here I think is to change this to an inference variable and then also check that we don't generate a requirement like `'?A: 'free`, but I'm not sure what the best way to do that is.